src/pudl/metadata/constants.py

```diff
@@ -14,6 +14,7 @@
     "date": "datetime64[ns]",
     "datetime": "datetime64[ns]",
     "year": "datetime64[ns]",
+    "category": "category",
```

Comment:
We might want to think about this a little bit in the context of how our resource definition & typing system works.

Right now we can set an `enum` constraint on a column, which results in a `CHECK VALUE IN () OR NULL` constraint being applied to the column when it's loaded into the database. How should that kind of constraint relate to the specification that a column is a `category` when it's loaded into a dataframe?